LCAS/limo_ros2
# Separate dev

## What type of PR is this? (check all applicable)


## Description

This now does the following:

* installs `zenoh-bridge-dds` (see https://zenoh.io/blog/2021-04-28-ros2-integration/) which allows to easily bridge the two ROS networks (using cyclone DDS, at least only tested with Cyclone!). E.g. running `zenoh-bridge-ros2dds -l tcp/0.0.0.0:8888` on the robot now starts a single TCP port server that any client can connect to via e.g `zenoh-bridge-ros2dds -e tcp/IP_OF_ROBOT:8888` and it makes all ROS DDS available on the client.
* configures the setup to use Cyclone DDS (was there before!), but now is configured to only work with the local address space and not use multicast (as we connect via Zenoh, see above).
* it now has 2 separate dev containers, on for development on the desktop (amd64) and one to work on the actual robot (arm64). They are called `on_amd64` and `on_limo` respectively. One should choose the right one depending on the situation (vscode prompts).
* correct the `devcontainer` format to comply with the standard, e.g. customisations
* it also installs a compiled version of the limo_ros2 repo into `/opt/ros/lcas` and configures its use, so this also now works without having to compile the workspace first (which can still be overlaid)

## Related Tickets & Documents



- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

Please try out the new `on_amd64` dev container, e.g. run
* `ros2 launch limo_gazebosim limo_gazebo_diff.launch.py`
* `zenoh-bridge-ros2dds -l tcp/0.0.0.0:8888`

and confirm that working. Optionally, try on the real platform

## [optional] Are there any post deployment tasks we need to perform?
